### DISCUSSION TITLE
One place for opencl variables

### DISCUSSION THREAD

**Discussion by lucas-wilkins** [SasView/sasview#2341] [General]
This should either be in the config, or in the environment variable, not both.

**Comment by llimeht** [comment]
Environment variables are very handy for automation such as turning opencl on/off for testing purposes, or configuring it on a cluster. Doing those things via a config file is awkward.

Config files are good for desktop users who want persistent settings. Environment variables tend to be quite awkward for that.

Both have their uses, but there needs to be a clear understanding of the precedence and only one place where that logic is coded. That place typically should be as close to the boundary of the program at possible and not at the point of use -- just like dealing with encoding/decoding unicode/bytes. That's not the current pattern, however, which is the source of pain.

**Comment by pkienzle** [comment]
Discussion #2226. I moved handling of SAS_OPENCL to `sas/system/lib.py` as part of #2280 so that GUI and script would use the same logic.
